doc/psyad/user_guide/implementation.rst

```diff
@@ -376,25 +376,31 @@ active then the loop statement is considered to be active. In this case:
           therefore avoid generating any loop-bound offset code in
           this case.
 
+Pre-processing
+++++++++++++++
+
+PSyAD implements an internal pre-processing phase where code
+containing unsupported code structures or constructs is transformed
+into code that can be processed. These structures/constructs are
+detailed below.
+
 Array Notation
 --------------
 
 Array notation in tangent-linear codes is translated into equivalent
-loops before the tangent-linear code is transformed into its
-adjoint. This is performed as the rules that are applied to transform
-a tangent-linear code into its adjoint are not always correct when
-array notation is used.
-
-.. note:: At the moment all array notation is translated into
-	  equivalent loops irrespective of whether the associated
-	  variables are active or not.
+loops in the pre-processing phase before the tangent-linear code is
+transformed into its adjoint. This is performed as the rules that are
+applied to transform a tangent-linear code into its adjoint are not
+always correct when array notation is used. Only array notation that
+contains active variables is translated into equivalent loops.
 
 Intrinsics
 ----------
 
 If an intrinsic function, such as ``matmul`` or ``transpose``, is
 found in a tangent-linear code and it contains active variables then
-it must be transformed to its associated adjoint form.
+it must be transformed to its associated adjoint form. This is
```

Comment:
"...transformed such that it is replaced by equivalent Fortran code. This is..."

Reply:
Modified.